### PR TITLE
Add Discord automated event reminder example

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
$subject. 

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: examples/automated-event-reminders/main.bal [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        return;
+    }
+
+    // Schedules the daily recurring task to send the event reminders to all the related channels.
+    task:JobId _ = check task:scheduleJobRecurByFrequency(new EventReminderJob(eventResponse.id), 60 * 60 * 24);
+}
+
+# A Ballerina scheduled job that sends a given event reminder message to a list of Discord channels.
+class EventReminderJob {
+
```
Comment (ayeshLK) [inline examples/automated-event-reminders/main.bal]:
```suggestion
```

File: examples/automated-event-reminders/main.bal [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    *task:Job;
+
+    string eventName;
```
Comment (ayeshLK) [inline examples/automated-event-reminders/main.bal]:
```suggestion
    private final string eventName;
```

File: examples/automated-event-reminders/main.bal [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    public function execute() {
+        anydata[]|error channelsResponse = discord->/guilds/[guildId]/channels();
+        if channelsResponse is error {
+            log:printError("Error getting the channels: ", channelsResponse);
+            return;
+        }
+
+        Channel[]|error channelsResult = <Channel[]>channelsResponse;
```
Comment (ayeshLK) [inline examples/automated-event-reminders/main.bal]:
```suggestion
        Channel[]|error channelsResult = channelsResponse.ensureType();
```

File: examples/automated-event-reminders/main.bal [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        clientId,
+        clientSecret,
+        scopes
+    }
+};
+
+discord:Client discord = check new (discordConfig);
+
+public function main() returns error? {
+
```
Comment (ayeshLK) [inline examples/automated-event-reminders/main.bal]:
```suggestion
```

File: examples/automated-event-reminders/main.bal [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        scheduled_end_time: "2024-12-31T23:59:59Z",
+        privacy_level: 2,
+        entity_type: 1,
+        entity_metadata: {
+            location: "Virtual"
+        }
+    };
+
+    Event|error eventResponse = discord->/guilds/[guildId]/scheduled\-events.post(payload);
+    if (eventResponse is error) {
```
Comment (ayeshLK) [inline examples/automated-event-reminders/main.bal]:
```suggestion
    if eventResponse is error {
```

File: examples/automated-event-reminders/main.bal [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        entity_metadata: {
+            location: "Virtual"
+        }
+    };
+
+    Event|error eventResponse = discord->/guilds/[guildId]/scheduled\-events.post(payload);
+    if eventResponse is error {
+        log:printError("Error creating the scheduled event: ", eventResponse);
+        return;
+    }
```
Comment (ayeshLK) [inline examples/automated-event-reminders/main.bal:69]: Can we do this ? I think here our focus is to show the usability of the discord connector. I think we can omit error handling part in this context.
```suggestion
    Event eventResponse = check discord->/guilds/[guildId]/scheduled\-events.post(payload);
```

Comment (NipunaRanasinghe) [reply examples/automated-event-reminders/main.bal:69]: Yeah thats also an option. But at the same time these examples should represent how these connectors **should** be used in real world scenarios, isn't it? So I thought including proper error handling logic will guide the new users who wants to use the connector to follow the best practices. WDYT?

Comment (ayeshLK) [reply examples/automated-event-reminders/main.bal:69]: In that case lets keep this +1

**Review by ayeshLK — APPROVED** [review]
LGTM

**Pull request merged**